### ISSUE TITLE
alpha blending issue for layers?

### ISSUE THREAD

**Issue by OlaKenji** [MarkelZ/pygame-render#7]
Hi! @MarkelZ 

I have notised that you get different results (in this case for a blur shader) if you blur directly onto the screen, or if you first blur it onto a layer, save it and then render this layer.texture. The difference in this case is you get a black outline or not (see attached figures). I have attached a simple script to reproduce these results (change blur_init to False or True). 

Not really sure why. Could it be that alpha blending for layers work a little different?

[blur.zip](https://github.com/MarkelZ/pygame-render/files/14802850/blur.zip)

<img width="1280" alt="True" src="https://github.com/MarkelZ/pygame-render/assets/65251858/cd27a05e-23a9-4c3b-b6c2-efda1495924b">
 
<img width="1278" alt="False" src="https://github.com/MarkelZ/pygame-render/assets/65251858/f4981799-3d3a-400b-85a5-dfebb2e277f5">

**Comment by MarkelZ** [comment]
Hey! @OlaKenji 

Thanks for your feedback. You're right, at first I thought it was the issue mentioned [here](https://github.com/MarkelZ/pygame-render/issues/4#issuecomment-1874246956), but then I saw that your blur shader actually implements that solution, while you still get a black outline.

Indeed, this is happening because when you render the blurred texture onto the empty layer, the engine alpha blends these two, whereas you just want to override the empty layer with the blurred texture. 

So, I suggest to disable alpha blending, check out this fix:

[blur_fixed.zip](https://github.com/MarkelZ/pygame-render/files/14811954/blur_fixed.zip)

I made two changes: 
1. I added the function `use_alpha_blending` to the `RenderEngine` class.
2. I do `engine.use_alpha_blending(False)` before rendering the blurred texture, and `engine.use_alpha_blending(True)` afterwards.

Does this look good to you? 

I think I can push these changes to the main branch since it doesn't affect backwards compatibility and it's a nice feature to have.
